mm2src/lp_ordermatch.rs

```diff
@@ -1986,13 +1984,10 @@ fn order_pair_root_mut<'a>(state: &'a mut HashMap<AlbOrderedOrderbookPair, H64>,
 }
 
 fn pair_history_mut<'a>(
-    state: &'a mut HashMap<AlbOrderedOrderbookPair, TrieOrderHistory>,
+    state: &'a mut TimeCache<AlbOrderedOrderbookPair, TrieOrderHistory>,
     pair: &str,
 ) -> &'a mut TrieOrderHistory {
-    match state.raw_entry_mut().from_key(pair) {
-        RawEntryMut::Occupied(e) => e.into_mut(),
-        RawEntryMut::Vacant(e) => e.insert(pair.to_owned(), Default::default()).1,
-    }
+    state.entry(pair.into()).or_insert_with(TrieOrderHistory::default)
```

Comment:
I've noticed that `or_insert_with` doesn't update the expiration of the entry if it's occupied. So even if it's used a lot it will be removed after TRIE_STATE_HISTORY_TIMEOUT. It is also undesirable to always update it in `or_insert_with`. I propose adding `or_insert_with_update_expiration` and use it here.

Alternatively, we can avoid it by keeping the `HashMap` for `AlbOrderedOrderbookPair -> TrieOrderHistory`, and adding TimeCache for TrieDiffHistory as proposed above.